scripts/format.sh

```diff
@@ -13,11 +13,13 @@
 # This script will use clang-tidy and clang-format to format code.
 #
 # Usage:
-#    ./format.sh <path to directory to format>
+#    ./format.sh <paths to directories or files to format>
 # 
 # We require clang-format 17.0.4. Other versions may produce different styles!
 # ---------------------------------------------------------------------------------
 
-find $1 -iname '*.h' -o -iname '*.hpp' -o \
+paths="$@"
```

Comment:
In `tidy_and_format.sh` if you reverse the input order so it's build path then directories format that would make it easier to pass multiple path to that script as well if that's helpful.

Reply:
I actually am going to remove `tidy_and_format.sh` because the clang-tidy part doesn't work too well (I had to do a lot of extra stuff to make it work). The easier way to apply clang-tidy is by enabling it in CMake (in the formatting PR I added this option too).